libyul/optimiser/LoadResolver.cpp

```diff
@@ -46,13 +51,31 @@ void LoadResolver::visit(Expression& _e)
 {
 	DataFlowAnalyzer::visit(_e);
 
+	auto isKeccak = [&](YulString const& _name) -> bool
+	{
+		if (auto evmDialect = dynamic_cast<EVMDialect const*>(&m_dialect))
+			if (auto builtin = evmDialect->builtin(_name))
+				return (
+					builtin->instruction &&
+					*builtin->instruction == evmasm::Instruction::KECCAK256
+				);
+
+		return false;
+	};
+
+
 	if (FunctionCall const* funCall = std::get_if<FunctionCall>(&_e))
+	{
 		for (auto location: { StoreLoadLocation::Memory, StoreLoadLocation::Storage })
 			if (funCall->functionName.name == m_loadFunctionName[static_cast<unsigned>(location)])
 			{
 				tryResolve(_e, location, funCall->arguments);
 				break;
 			}
+
+		if (isKeccak(funCall->functionName.name))
```

Comment:
Should we add some checks for `runs` parameter here?

Reply:
Yes, please, the big constants are rather expensive.

Reply:
Added a check based on optimize-runs to see if the rule should be applied.